File: DataStructureAlgorithm/BinarySearch/binarySearch.py
def BinarySearch(arr, value):
    left, mid, right, count = 0, 0, len(arr)-1, 0
    print("value({})-> ".format(value), end="")
    while left<=right:
        count+=1
        mid = int(left+(right-left)/2)
        if arr[mid]==value:
           print("ok, count: %d, " % count, end="")
           return mid
        elif arr[mid] < value:
            left = mid+1
        else:
            right = mid-1
    print("fail, count: %d, " % count, end="")
    return -1

if __name__=="__main__":
    a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
    for i in range(1, 12):
        r = BinarySearch(a, i)
        print("value index:", r)

File: DataStructureAlgorithm/BinarySearch/binary_search.rs
fn main(){
    let r = [1,2,3,4,5,6,7,8];
    for i in 1..=8{
        let vi = binary_search(&r, i);
        if let None = vi{
            break;
        }else{
            println!("value index: {}", vi.unwrap());
        }
    }
}

fn binary_search(arr: &[i32], value: i32)->Option<usize>{
    let size = arr.len();
    let mut count = 0;
    let mut lmr:(usize, usize, usize) = (0,0,size-1);//left, middle, right
    print!("value({})=> ", value);
    while lmr.0<=lmr.2{
        count+=1;
        lmr.1 = lmr.0+(lmr.2-lmr.0)/2;
        if arr[lmr.1]==value{
            print!("ok, count: {}, ", count);
            return Some(lmr.1);
        }else if arr[lmr.1]<value{
            lmr.0 = lmr.1+1;
        }else{
            lmr.2 = lmr.1-1;
        }
    }
    print!("fail, count: {}, ", count);
    None
}